parking_lot.py

```diff
@@ -0,0 +1,172 @@
+from datetime import datetime
+from typing import Optional, Union
+from enum import Enum
+from pydantic import BaseModel, Field, ValidationError, validator
+from fastapi import FastAPI, Request, status
+from starlette.responses import JSONResponse
+import math
+import dateutil.parser
+
+app = FastAPI()
+
+PARKED_CARS = {
+    'X773HY97': {'status': 'parked', 'tariff': 'hourly', 'location': 7, 'start': '2022-06-25T17:21:33.913233', 'end': None, 'fee': None},
+    'G737TT97': {'status': 'parked', 'tariff': 'daily', 'location': 14, 'start': '2022-06-25T17:23:24.428268', 'end': None, 'fee': None}
+}
```

Comment:
What is required in your program to don't blow up the memory if we have > 1,000,000 items ?

Reply:
Let's say model is greater in size in the future, memory will increase over time since this is the data store mechanism at the moment

Reply:
this is a simulation representing the database since the requirements specified that a database was not needed.

addressing such a problem would be to make use of a database